### PR TITLE
feat(reporter): Only include specified categories in the sources bundle

### PR DESCRIPTION
**Pull request by MarcelBochtler**
Similar to the ORT CLI [1], the SourceCode Bundle should only include packages from specified license categories.

[1]: https://github.com/oss-review-toolkit/ort/blob/8181666dc5d8bfd739b1260c171c4e707c0c7568/model/src/main/resources/reference.yml#L163-L165

### CODE REVIEW

File: workers/reporter/src/main/kotlin/reporter/SourceCodeBundleReporter.kt [hunk 1]
```diff
@@ -162,4 +192,25 @@ class SourceCodeBundleReporter(
...
+        val resolvedLicenseInfo = licenseInfoResolver.resolveLicenseInfo(id)
+        val effectiveLicenses = resolvedLicenseInfo.effectiveLicense(
+            LicenseView.CONCLUDED_OR_DECLARED_AND_DETECTED,
+            licenseChoices
+        )?.decompose().orEmpty()
+
+        return licenseCategorizations
+            .filter { it.id in effectiveLicenses }
+            .flatMap { it.categories }
+            .toSet()
```
Comment (sschuberth) [inline workers/reporter/src/main/kotlin/reporter/SourceCodeBundleReporter.kt:214]: See https://github.com/oss-review-toolkit/ort/pull/8654 😉

File: workers/reporter/src/main/kotlin/reporter/SourceCodeBundleReporter.kt [hunk 2]
```diff
@@ -92,6 +112,16 @@ class SourceCodeBundleReporter(
                 log.info("Found ${packages.size} packages(s) in the ORT result.")
                 addAll(packages)
             }
+        }.filter { pkg ->
+            if (includedLicenseCategories.isEmpty()) { return@filter true }
```
Comment (sschuberth) [inline workers/reporter/src/main/kotlin/reporter/SourceCodeBundleReporter.kt:116]: Consider only filtering at all if `includedLicenseCategories` is non-empty, like

        val filteredPackages = packages.takeIf { includedLicenseCategories.isEmpty() } ?: packages.filter { pkg ->
            val pkgLicenseCategories = pkg.getLicenseCategories(
                licenseClassifications.categorizations,
                ortResult.createLicenseInfoResolver(),
                ortResult.getPackageLicenseChoices(pkg.id)
            )

            pkgLicenseCategories.any { it in includedLicenseCategories }
        }

and then using `filteredPackages` below.

Comment (MarcelBochtler) [reply workers/reporter/src/main/kotlin/reporter/SourceCodeBundleReporter.kt:116]: While it is obviously better to not applying the filter if there is nothing to filter, I don't like that your approach would introduce a new variable where the old one (`packages`), should not be used in the code below.

File: workers/reporter/src/main/kotlin/reporter/SourceCodeBundleReporter.kt [hunk 3]
```diff
@@ -94,13 +114,23 @@ class SourceCodeBundleReporter(
             }
         }
 
+        val filteredPackages = allPackages.takeIf { includedLicenseCategories.isEmpty() } ?: allPackages.filter { pkg ->
+            val pkgLicenseCategories = pkg.getLicenseCategories(
+                licenseClassifications.categorizations,
+                ortResult.createLicenseInfoResolver(),
```
Comment (oheger-bosch) [inline workers/reporter/src/main/kotlin/reporter/SourceCodeBundleReporter.kt:120]: Not sure what exactly `ortResult.createLicenseInfoResolver()` does, but I assume such a resolver should be created only once and not for each element to be filtered.

**Review by oheger-bosch — APPROVED** [review]

**Pull request merged**